src/compute-types/src/plan/render_plan.rs

```diff
@@ -697,24 +654,81 @@ impl<T: Clone> FlatPlan<T> {
             return vec![self];
         }
 
+        let bodies = self.body.partition_among(parts);
```

Comment:
Took me a minute to understand why we process the bodies first. Might pay to leave a comment?